packages/flutter_tools/lib/src/flutter_manifest.dart

```diff
@@ -51,10 +51,17 @@ class FlutterManifest {
   /// Returns null on missing or invalid manifest.
   @visibleForTesting
   static FlutterManifest? createFromString(String manifest, {required Logger logger}) {
-    return _createFromYaml(loadYaml(manifest), logger);
+    try {
+      return createFromYaml(loadYaml(manifest), logger);
+    } on FormatException catch (e) {
+      logger.printStatus('Malformed pubspec.yaml:', emphasis: true);
+      logger.printError(e.message);
+      return null;
+    }
   }
 
-  static FlutterManifest? _createFromYaml(Object? yamlDocument, Logger logger) {
+  /// Returns null on missing or invalid manifest.
```

Comment:
Now that this is a public API: please expand on the comment and explain what exactly this does.

In particular, it would also be nice to explain what `yamlDocument` is expected to be as its a spooky `Object?` type (i.e. anything). Is that still necessary?